src/Storages/StoragePostgreSQL.cpp

```diff
@@ -123,6 +124,10 @@ Pipe StoragePostgreSQL::read(
         column_names_,
         storage_snapshot->metadata->getColumns().getOrdinary(),
         IdentifierQuotingStyle::DoubleQuotes, remote_table_schema, remote_table_name, context_);
+
+    /// Single quotes in PostgreSQL are escaped through repetition
+    boost::replace_all(query, "\\'", "''");
```

Comment:
It does not work for this ClickHouse query:

```
SELECT '\\\''
```

it will translate it to the following query:
```
SELECT '\\'''
```

while Postgres will expect the following query:
```
SELECT '\'''
```